tests/color_format.rs

```diff
@@ -0,0 +1,137 @@
+#![cfg(feature = "fancy-no-backtrace")]
+
+use lazy_static::lazy_static;
+use miette::{Diagnostic, MietteHandler, MietteHandlerOpts, ReportHandler, RgbColors};
+use regex::Regex;
+use std::fmt::{self, Debug};
+use std::sync::Mutex;
+use thiserror::Error;
+
+#[derive(Eq, PartialEq, Debug)]
+enum ColorFormat {
+    NoColor,
+    Ansi,
+    Rgb,
+}
+
+#[derive(Debug, Diagnostic, Error)]
+#[error("oops!")]
+#[diagnostic(code(oops::my::bad), help("try doing it better next time?"))]
+struct MyBad;
+
+struct FormatTester(MietteHandler);
```

Comment:
Is there a better way to do this? Not super familiar with miette's API, and there might be something I missed.

Reply:
https://github.com/zkat/miette/blob/main/tests/graphical.rs#L9-L35 this is how the current tests do this, if I understand your question.

Reply:
Yeah, that would be the equivalent for the other tests. Unfortunately, calling `render_report` directly doesn't work here because the automatic color selection happens as part of `MietteHandler`, which just implements the `ReportHandler` trait, and doesn't have an equivalent to `render_report`. Gonna leave the `FormatTester` struct as-is.